plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/ImplementCovariancePop.java

```diff
@@ -0,0 +1,77 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.expression;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcExpression;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.Type;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.base.Verify.verifyNotNull;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.expressionTypes;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.inputs;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.variables;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.RealType.REAL;
+import static java.lang.String.format;
+
+public class ImplementCovariancePop
+        implements AggregateFunctionRule
+{
+    private static final Capture<List<Variable>> INPUTS = newCapture();
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().matching(name -> name.equalsIgnoreCase("covar_pop")))
+                .with(inputs().matching(
+                        variables()
+                                .matching(expressionTypes(REAL, REAL).or(expressionTypes(DOUBLE, DOUBLE)))
+                                .capturedAs(INPUTS)));
+    }
+
+    @Override
+    public Optional<JdbcExpression> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context)
+    {
+        List<Variable> inputs = captures.get(INPUTS);
+        verify(inputs.size() == 2);
+
+        JdbcColumnHandle columnHandle1 = (JdbcColumnHandle) context.getAssignments().get(inputs.get(0).getName());
+        verifyNotNull(columnHandle1, "Unbound variable: %s", inputs.get(0));
+
+        JdbcColumnHandle columnHandle2 = (JdbcColumnHandle) context.getAssignments().get(inputs.get(1).getName());
+        verifyNotNull(columnHandle2, "Unbound variable: %s", inputs.get(1));
+
+        Set<Type> types = ImmutableSet.of(aggregateFunction.getOutputType(), columnHandle1.getColumnType(), columnHandle2.getColumnType());
+        verify(types.equals(ImmutableSet.of(REAL)) || types.equals(ImmutableSet.of(DOUBLE)));
```

Comment:
You know that `columnHandle1.getColumnType(), columnHandle2.getColumnType()` are the same, so it's an indirect way of saying "verify outputtype  equals columnHandle1.getColumnType()"
let's make it explicit